docs/source/design/load.rst

```diff
@@ -27,56 +27,51 @@ Each builder class implements methods for the following behaviors:
-   previous steps are passed to this method as well so that they can be
-   included as part of the payload if desired.
+#. **Compose the entity's uniquely identifying primary elements**. This is used
+   for querying identifiers assigned by the target service so that they can be
+   repeatably assigned to the same entities again. (Target services often don't
+   do this equivalent entity replacement themselves.)
+#. **Build a complete message payload that will be sent to the server to
+   populate a new (or updated) entity of its type**.
 
 For more details, read :ref:`Tutorial-Make-Target-Service-Plugin`
 
 ----
 
 **A Note on the ID Cache**
 
-It may be easier to understand the purpose of the ID cache and
-unique keys with an example.
+It may be easier to understand the purpose of the unique key components with an
+example.
 
 Given the record:
 
 .. csv-table::
-   :header: PARTICIPANT.ID, PHENOTYPE.NAME, PHENOTYPE.EVENT_AGE_DAYS,\
-            PHENOTYPE.OBSERVED
-
-   P1,  Cleft Ear, 765, Positive
-
-The unique keystring for the phenotype observation in this record could be:
-``"p1__cleft-ear__765"``. (Assume that participant P1 is not Schrödinger's cat
-and can only be either positive or negative for that characteristic at that
-time, but not both simultaneously.)
-
-And, after submission to the target service, the returned identifier for the
-phenotype observation entity might be something like: ``"PH_00001111"``.
-
-The ingest library will internally create an entry in an ID cache for the
-current target service URL which associates this unique key with its target
-identifier like this:
-
-+------------------------------------+
-| phenotypes                         |
-+--------------------+---------------+
-| Unique Key         | Target ID     |
-+====================+===============+
-| p1__cleft-ear__765 |  PH_00001111  |
-+--------------------+---------------+
-
-Then whenever the ingest library sees a record in the source data with the
-unique key components above, it will always associate the resulting phenotype
-observation with the unique keystring ``"p1__cleft-ear__765"`` which will
-always point to the phenotype observation entity in the target service
-identified by ``"PH_00001111"``.
+   :header: STUDY.TARGET_SERVICE_ID, PARTICIPANT.ID, PARTICIPANT.AGE, PARTICIPANT.SEX
+
+   SD_12345678,  P1, 17, Male
+
+The uniquely identifying components for the participant in this record would
+be: ``{"study_id": "SD_12345678", "external_id": "P1"}`` because every
+participant external ID is guaranteed to be unique only within a given study.
+Any future corrections to the participants age or sex should be associated with
+the same entry.
+
+Let's say that, after submission to the target service, the returned identifier
+for the participant entity is something like: ``"PT_00001111"``.
+
+The ingest library will internally create an association relating this unique
+key with its target identifier like this:
```

Comment:
Isn't what creates the internal association a target API builder (ingest plugin) with the ingest library invoking this builder? Since we imagine ingest plugins against other target services are developed separately from the ingest library, why don't we clarify that a little bit here?

Reply:
The api builder returns the important parts, but the library itself contains the cache relation.